DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorRecHitsSoABase.cc

```diff
@@ -0,0 +1,122 @@
+// -*- C++ -*-
+///bookLayer
+// Package:    SiPixelPhase1MonitorRecHitsSoA
+// Class:      SiPixelPhase1MonitorRecHitsSoA
+//
+/**\class SiPixelPhase1MonitorRecHitsSoA SiPixelPhase1MonitorRecHitsSoABase 
+*/
+//
+// Author: Alessandro Rossi, Suvankar Roy Chowdhury
+//
+#include "DataFormats/Math/interface/approx_atan2.h"
+#include "CUDADataFormats/Common/interface/Product.h"
+#include "CUDADataFormats/TrackingRecHit/interface/TrackingRecHit2DHeterogeneous.h"
+// Geometry
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "SiPixelPhase1MonitorRecHitsSoABase.h"
+
+SiPixelPhase1MonitorRecHitsSoABase::SiPixelPhase1MonitorRecHitsSoABase(const edm::ParameterSet& iConfig)
+    : geomToken_(esConsumes<TrackerGeometry, TrackerDigiGeometryRecord, edm::Transition::BeginRun>()),
+      topoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd, edm::Transition::BeginRun>()) {
+  topFolderName_ = iConfig.getParameter<std::string>("TopFolderName");  //"SiPixelHeterogeneous/PixelRecHitsSoA";
+  onGPU_ = iConfig.getParameter<bool>("onGPU");                         //unused now, but may be needed
```

Comment:
I would prefer _not_ to have a parameter without any effect, because setting it to `True` or `False` will be misleading :-(

Reply:
Instead, better add it later, if it becomes needed.